src/Riok.Mapperly/Descriptors/MappingBodyBuilders/ObjectMemberMappingBodyBuilder.cs

```diff
@@ -135,6 +135,27 @@ PropertyMappingConfiguration config
             return false;
         }
 
+        // cannot assign to intermediate value type
+        if (targetMemberPath.Path.Count > 1)
+        {
+            var member = targetMemberPath.Path.Reverse().Skip(1).First();
```

Comment:
This only checks the second last item, but couldn't be any in the path except for the last item be a problem?
Otherwise this could be simplified to `var member = targetMemberPath.Path[^2];` / `var member = targetMemberPath.Path[targetMemberPath.Path.Count - 2];` (Index doesn't work due to net standard 2.0).